packages/frontend/src/ee/pages/AiAgents/AgentConversationPage.tsx

```diff
@@ -9,110 +13,174 @@ import {
-                            fz="lg"
-                            bg={
-                                humanScore === 1
-                                    ? 'green.7'
-                                    : humanScore === -1
-                                    ? 'red.7'
-                                    : undefined
-                            }
-                        >
-                            {humanScore === 1
-                                ? '👍'
-                                : humanScore === -1
-                                ? '👎'
-                                : undefined}
-                        </Badge>
-                    )}
-                </Card>
-            </Stack>
-        </Group>
+                    <Tooltip
+                        label={dayjs(message.createdAt).toString()}
+                        withinPortal
+                    >
+                        <Text size="xs" c="dimmed">
+                            {timeAgo}
+                        </Text>
+                    </Tooltip>
+
+                    <MessageBubble message={message}>
+                        {message ? (
+                            <MDEditor.Markdown
+                                source={message.message}
+                                style={{ backgroundColor: 'transparent' }}
+                            />
+                        ) : (
+                            <Text c="dimmed">No response yet</Text>
+                        )}
+
+                        {message.role === 'assistant' &&
+                            typeof message.humanScore === 'number' && (
+                                <Badge
+                                    pos="absolute"
+                                    right={10}
+                                    bottom={-14}
+                                    variant="filled"
+                                    size="lg"
+                                    fz="lg"
+                                    bg={
+                                        message.humanScore === 1
+                                            ? 'green.7'
+                                            : message.humanScore === -1
+                                            ? 'red.7'
+                                            : undefined
+                                    }
+                                >
+                                    {message.humanScore === 1
+                                        ? '👍'
+                                        : message.humanScore === -1
+                                        ? '👎'
+                                        : undefined}
+                                </Badge>
+                            )}
+                    </MessageBubble>
+                </Stack>
+            </Group>
+
+            {message.role === 'assistant' &&
+                message.vizConfigOutput &&
+                message.metricQuery && <AiResultMessage message={message} />}
+        </>
```

Comment:
I think you've adjusted `AiResultMessage` in a way that it can render both the text message + the viz in a single message bubble. we can peer on this while restacking on my PR